mender.go

```diff
@@ -26,50 +26,78 @@ import (
 )
 
 type Controler interface {
-	GetState() MenderState
+	Bootstrap() error
+	TransitionState() MenderState
 	GetCurrentImageID() string
 	GetDaemonConfig() daemonConfig
 	GetUpdaterConfig() httpsClientConfig
+	LastError() error
 }
 
 const (
 	defaultManifestFile = "/etc/build_mender"
+	defaultKeyFile      = "/data/mender/mender-agent.pem"
 )
 
 type MenderState int
 
+// State transitions:
+//
+//            unknown
+//               |
+//               v
+//             init
+//               |
+//               v
+//          bootstrapped
+//               |
+//       +-------+-------------+
+//       |                     |
+//       v                     v
+// fresh update         wait for update
```

Comment:
I think this might be bit more complicated than that. What if you have the update but it is failed one (aka some checks are failing). Then we will need to reboot the device and use previously active partition as the one you will boot from.